### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by GabyForceQ**

### CODE REVIEW

**Comment by Hackerpilot** [comment]
Title is "Code improvements", "12 files changed". Here we go....

File: src/containers/cyclicbuffer.d [hunk 1]
```diff
@@ -7,10 +7,12 @@
 
 module containers.cyclicbuffer;
 
-private import core.exception : onRangeError;
-private import stdx.allocator.mallocator : Mallocator;
-private import std.range.primitives : empty, front, back, popFront, popBack;
-private import containers.internal.node : shouldAddGCRange;
+@trusted:
```
Comment (Hackerpilot) [inline src/containers/cyclicbuffer.d]: Have you actually made sure that all the code in here is memory safe?

Comment (Hackerpilot) [reply src/containers/cyclicbuffer.d]: And if so, how?

**Comment by rikkimax** [comment]
The expression based contracts will place a hard requirement on 2.081 or newer, so that changes the libraries minimum dmd version.

**Comment by BBasile** [comment]
Next time try to make a PR that target a specific issues and not a bag of unrelated changes like here.
Also the way you used `@trusted` was really not acceptable. Closing since the review comments have not been addressed.

**Pull request closed without merging**